dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -221,249 +241,260 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
         }
     }
 
-    void PushNotificationManager::RegisterActivator(PushNotificationActivationInfo const& details)
+    bool PushNotificationManager::IsBackgroundTaskRegistered(winrt::hstring const& backgroundTaskFullName)
     {
-        THROW_HR_IF(E_NOTIMPL, !::Microsoft::Windows::PushNotifications::Feature_PushNotifications::IsEnabled());
+        auto tasks{ BackgroundTaskRegistration::AllTasks() };
+        bool isTaskRegistered{ std::any_of(std::begin(tasks), std::end(tasks),
+            [&](auto&& task)
+            {
+                auto name{ task.Value().Name() };
 
-        try
-        {
-            THROW_HR_IF_NULL(E_INVALIDARG, details);
+                if (name == backgroundTaskFullName)
+                {
+                    m_pushTriggerRegistration = task.Value();
+                    return true;
+                }
 
-            auto registrationActivators{ details.Activators() };
+                return false;
+            }) };
 
-            auto isBackgroundTaskFlagSet{ WI_IsAnyFlagSet(registrationActivators, PushNotificationRegistrationActivators::PushTrigger | PushNotificationRegistrationActivators::ComActivator) };
-            THROW_HR_IF(E_INVALIDARG, isBackgroundTaskFlagSet && !IsActivatorSupported(registrationActivators));
+        return isTaskRegistered;
+    }
 
-            auto isProtocolActivatorSet{ WI_IsFlagSet(registrationActivators, PushNotificationRegistrationActivators::ProtocolActivator) };
-            THROW_HR_IF(E_INVALIDARG, isProtocolActivatorSet && !IsActivatorSupported(registrationActivators));
+    void PushNotificationManager::Register()
+    {
+        static bool registering{ false };
+        {
+            auto lock{ m_lock.lock_exclusive() };
+            THROW_HR_IF_MSG(E_FAIL, registering, "Registration is in progress!");
+            registering = true;
+        }
 
-            if (isProtocolActivatorSet)
+        try
+        {
+            if (PushNotificationHelpers::IsPackagedAppScenario())
             {
+                auto scopeExitToCleanRegistrations{ wil::scope_exit([&]()
```

Comment:
You need a check here: If the app has already registered, simply throw an error. We shouldn't allow multiple registrations.